bot-nova.py

```diff
@@ -164,6 +179,44 @@ async def main():
             tools=tools
         )
 
+        # Create tool processor BEFORE callbacks (so callbacks can reference it)
+        tool_processor = ToolProcessor(auth_token=bearer_token, learning_context=learning_context)
+
+        # Register function callbacks with LLM service (inherited from LLMService)
+        async def learning_component_callback(function_name, tool_call_id, arguments, llm, context, result_callback):
```

Comment:
We shouldn't have "learning_component", "whiteboard" or "video" in the python. We need to make all these calls dynamic/generic.